### PR TITLE
Vendor gardener@1.71.0 and adapt changes for workerless `Shoot`s

### PR DESCRIPTION
**Pull request by acumino**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:
- Vendor gardener v1.70.2

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The following dependency has been updated:
  - github.com/gardener/gardener 1.67.1 -> 1.70.2
```

### CODE REVIEW

File: pkg/admission/mutator/webhook.go [hunk 1]
```diff
@@ -59,7 +61,8 @@ func createCiliumPredicate() predicate.Funcs {
 			return false
 		}
 
-		return shoot.Spec.Networking.Type == cilium.ReleaseName
+		return !gardencorev1beta1helper.IsWorkerless(shoot) && shoot.Spec.Networking != nil &&
```
Comment (ScheererJ) [inline pkg/admission/mutator/webhook.go:64]: Is this not a bit redundant? Why checking for whether the shoot is workerless when we check for the networking spec afterwards, which will most likely be `nil` for workerless shoots?
Is it just to highlight the case when the networking spec can be `nil` or did you have something else in mind?

**Review by ScheererJ — COMMENTED** [review]
Thanks for the change.

I just have one question.

File: pkg/admission/mutator/webhook.go (hunk 1 above)
Comment (acumino) [reply pkg/admission/mutator/webhook.go]: Adapted in 81fda70bd7b12a2716fb6fc53d9f869d3a82b034

**Review by DockToFuture — APPROVED** [review]
/lgtm

**Pull request merged**